src/app/server/Mdns.cpp

```diff
@@ -246,7 +241,18 @@ CHIP_ERROR Advertise(bool commissionableNode, CommissioningMode mode)
             advertiseParameters.SetPairingInstr(chip::Optional<const char *>::Value(pairingInst));
         }
     }
+    return CHIP_NO_ERROR;
+}
 
+/// Overloaded utility method for commissioner and commissionable advertisement
+/// This method is used for both commissioner discovery and commissionable node discovery since
+/// they share many fields.
+///   commissionableNode = true : advertise commissionable node
+///   commissionableNode = false : advertise commissioner
+CHIP_ERROR Advertise(bool commissionableNode, CommissioningMode mode)
+{
+    auto advertiseParameters = chip::Mdns::CommissionAdvertisingParameters();
```

Comment:
why the auto? Why not:

```cpp
chip::Mdns::CommissionAdvertisingParameters advertiseParameters;
```

otherwise this seems to make it seem as you are calling a method with some unknown parameters.

The method seems to create a variable on the stack and then call build on it, never using it again. Why is that? if the argument is never used, why do we need to pass in a reference?